### PR TITLE
adding cdi-alternative files round 2

### PR DESCRIPTION
**Pull request by nzhu972**
I created a new branch to submit the cdi-alternative quickstart example because there are problems on the other branch which is preventing me from pushing the changes.

The project is structured in a single WAR file and followed the standards outlined in the jdf contribution page.

### CODE REVIEW

File: cdi-alternative/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+When more than one version of a bean is implemented for different purposes, the ability to switch between the versions during the development phase by injecting one qualifier or another is shown in this demo.
+
+Instead of having to change the source code of the application, one can make the choice at deployment time by using alternatives.
+
+Alternatives are commonly used for purposes like the following:
+
+1)  To handle client-specific business logic that is determined at runtime
+2)  To specify beans that are valid for a particular deployment scenario (for example, when country-specific sales tax laws require country-specific sales tax business logic)
+3)  To create dummy (mock) versions of beans to be used for testing
```
Comment (sgilda) [inline cdi-alternative/README.md]: These don't have line breaks for the numbered list when rendered in HTML. Try this with only 1 space between the period after the number and the start of the sentence. I would also get rid of the parentheses. 
1. To handle client-specific business logic that is determined at runtime.
2. To specify beans that are valid for a particular deployment scenario, for example, when country-specific sales tax laws require country-specific sales tax business logic.
3. To create dummy  or mock versions of beans to be used for testing.

File: cdi-alternative/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+        mvn clean package jboss-as:deploy
+        
+4. This will deploy `target/ear/cdi-alternative.ear` to the running instance of the server.
+
+
+Access the application
+---------------------
+
+The application will be running at the following URL <http://localhost:8080/cdi-alternative/Demo>.
```
Comment (sgilda) [inline cdi-alternative/README.md]: Most of the other quickstarts are accessed using the base URL, for example, http://localhost:8080/cdi-alternative/. That gives a 404 NOT FOUND. Is it possible to redirect it to the Demo page?

File: cdi-alternative/README.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+A message will display for the bean being injected during run time. 
+
+The alternative version of the beans can be updated in /WEB-INF/beans.xml by removing the <alternative> tag 
+
+or 
+
+change the class name to a different alternative.
```
Comment (sgilda) [inline cdi-alternative/README.md]: This is hard to read. Maybe something like this:

You can specify alternative versions of the bean in the WEB-INF/beans.xml file by doing one of the following:
1. You can remove the `<alternatives>` tag.
2. You can change the class name.

File: cdi-alternative/README.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+The alternative version of the beans can be updated in /WEB-INF/beans.xml by removing the <alternative> tag 
+
+or 
+
+change the class name to a different alternative. 
+
+In this quickstart, in order to switch back to the default implementation, 
+comment the <alternative> block. 
+
```
Comment (sgilda) [inline cdi-alternative/README.md:75]: I think you have to put tick marks around the `<alternatives>` tag. It's plural, plus, it doesn't appear in the text. :-) 
You also need to redeploy the application. 

To switch to the default implementation, comment the `<alternatives>` block in the WEB-INF/beans.xml file and redeploy the quickstart.

**Comment by sgilda** [comment]
Tested this and it works great! Other than the few minor comments above, I'm good with this one.

@pmuir or @LightGuard : Do you want to review the code?

File: cdi-alternative/src/main/java/org/jboss/as/quickstarts/cdi/alternative/Demo.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+     * If beans.xml has alternative defined, then that class will be injected.  
+     * Otherwise, the default is injected.
+	 */
+    @Inject
+    private Tax tax;       
+    /**
+     * default constructor
+     */
+    public Demo() {
+        super();
```
Comment (LightGuard) [inline cdi-alternative/src/main/java/org/jboss/as/quickstarts/cdi/alternative/Demo.java]: Not that the ctor hurts anything, but it certainly isn't needed.

File: cdi-alternative/README.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+Instead of having to change the source code of the application, one can make the choice at deployment time by using alternatives.
+
+Alternatives are commonly used for purposes like the following:
+
+1)  To handle client-specific business logic that is determined at runtime
+2)  To specify beans that are valid for a particular deployment scenario (for example, when country-specific sales tax laws require country-specific sales tax business logic)
+3)  To create dummy (mock) versions of beans to be used for testing
+
+To make a bean available for lookup, injection, or EL resolution using this mechanism, give it a javax.enterprise.inject.Alternative annotation and then use the alternative element to specify it in the beans.xml file.
```
Comment (LightGuard) [inline cdi-alternative/README.md]: This isn't really accurate. Any Java class which has a no-args constructor and is in an archive with a beans.xml is available for lookup and injection. For EL resolution it must contain `@Named`

**Comment by LightGuard** [comment]
I added a couple of comments.

**Comment by nzhu972** [comment]
I updated README and the servlet based on both of your feedback. The page can also be accessed via http://localhost:8080/cdi-alternative as suggested instead of /Demo.

**Comment by sgilda** [comment]
Thanks!. I'll try to look at it tomorrow.

**Comment by sgilda** [comment]
I just realized the README is missing the `Target Product` metadata tag, however. I can add it later when I fix some other things. or before I merge.

Looks good to me.

@LightGuard : Let me know if you're OK with this and I'll merge it.

**Comment by LightGuard** [comment]
Looks fine to me

**Comment by sgilda** [comment]
Merged upstream.

**Pull request closed without merging**